src/Io/Transaction.php

```diff
@@ -35,8 +36,19 @@ class Transaction
 
     private $streaming = false;
 
-    public function __construct(RequestInterface $request, Sender $sender, array $options = array(), MessageFactory $messageFactory)
-    {
+    /** @var int $timeout */
+    private $timeout = 10;
```

Comment:
Not sure about this default value, how about using http://php.net/manual/en/filesystem.configuration.php#ini.default-socket-timeout instead? (defaults to 60s)